app/models/piece.rb

```diff
@@ -63,4 +63,25 @@ def is_obstructed?(start_point, end_point)
 		end
 	end
 
+	#get piece's current (start) position
+	def current_pos
+		return arr[pos_x, pos_y]
+	end
+
+	#check if a move is being made to a new position
+	def new_point?(end_point)
+		end_x, end_y = end_point
+		((pos_x != end_x) || (pos_y != end_y)) ? true : false
```

Comment:
this would be easier if written as just

!(pos_x == end_x && pos_y == end_y)

The or condition is harder to reason about at a quick glance in my opinion but I guess that's just a preference.  Really though you don't have to explicitly return true or false because the result of 

!(pos_x == end_x && pos_y == end_y) is either true or false so the true false you have is redundant.